### DISCUSSION TITLE
REGEX YouTube Regel

### DISCUSSION THREAD

**Discussion by Chris56564** [AdguardTeam/AdGuardHome#5105] [General]
Hello, I'm trying to create a REGEX rule, but unfortunately it doesn't work.

I will block
rr4---sn-4g5edndy.googlevideo.com
rr1---sn-4g5ednds.googlevideo.com
rr2---sn-4g5ednsd.googlevideo.com
rr4---sn-4g5lznls.googlevideo.com
.........
My Rule is
/REGEX/rr\d+-+sn-\d\w\d\w\w\w\w\w\.googlevideo\.com
or
/REGEX/rr[0-9]---sn-[a-z0-9]{8}\.googlevideo\.com

What am I doing wrong?

Chris

**Comment by MkQtS** [comment]
The syntax is wrong. You may see the [wiki](https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#regular-expressions).

The rule in your case should be:

```
/rr\d+-+sn-\d\w\d\w\w\w\w\w.googlevideo.com/
```

or

```
/rr[0-9]---sn-[a-z0-9]{8}.googlevideo.com/
```

However, if you want to block these domains, `||googlevideo.com^` is enough.

**Comment by Concubine** [comment]
Hey peeps, in relation to this thing:

||^rr[[:digit:]]{1,}---[[:alnum:]]{1,}-[[:alnum:]]{1,}-[[:alnum:]]{1,}[[:alnum:]]{1,}\.googlevideo\.com$important

This should match any variation but I was wondering if the regex rules support anchors?

**Reply by Concubine** [reply]
UPDATE: ^rr[[:digit:]]{1,}---[[:alnum:]]{1,}-[[:alnum:]]{1,}.[[:alnum:]]{1,}[[:alnum:]]{1,}\.googlevideo\.com$

the last catch googlevideo.com$ works brilliantly.. hahahah just wondering why the above isn't catching.

Also the googlevideo.com kinda breaks the flow of watching stuff, I think youtube are detecting it and it is kinda annoying them or these are actual video links as well and not only ads
